packages/core/src/browser/tree/tree-widget.ts

```diff
@@ -125,8 +125,16 @@ export class TreeWidget extends VirtualWidget implements StatefulWidget {
 
     protected onActivateRequest(msg: Message): void {
         super.onActivateRequest(msg);
-        if (!this.model.selectedNodes && SelectableTreeNode.is(this.model.root)) {
-            this.model.selectNode(this.model.root);
+        if (this.model.selectedNodes.length === 0) {
```

Comment:
it should skip invisible nodes, you need something like `TreeModelImpl.selectNextVisibleNode`

Reply:
Shouldn't "selectable" imply "visible"?